keps/sig-testing/5196-invariants/README.md

```diff
@@ -0,0 +1,119 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This proposal defines a system to gather and analyze invariant signals about the Kubernetes cluster during test execution. We want to see how key parts of the system are behaving in a way that's similar to how real users would see it. This will help us find problems that normal tests might miss, and give us a better picture of how stable and reliable Kubernetes is.
+
+## Motivation
+
+Even with extensive testing, it's impossible to catch every potential problem in Kubernetes. Some issues only show up when the system is used in real-world scenarios. We want to use our testing setup to act like a group of users, watching for critical signs that something might be wrong. These signs, or "invariants," can tell us if the cluster is healthy and working as expected. By collecting and analyzing this information, we can find and fix problems before they affect real users.
+
+### Goals
+
+- To build a system that collects and reports on how well defined invariants are behaving during our tests.
+- To get a better understanding of how these invariants perform in a situation that looks like real user usage, so we can find potential problems early.
+- To create a way to look at and understand this invariant information, so we can make better decisions about the system.
+
+### Non-Goals
+
+- To make invariant checks part of the regular pass/fail tests.
+- To change the way our tests work now.
+- To collect invariant data from places that normal users can't easily see.
+- To build a complex system; we'll focus on collecting and looking at the information first.
+
+## Proposal
+
+We will add to our current testing system so it can collect and report invariant data during tests. This data will be stored in a central place where we can look at it later. We will use existing tools whenever possible.
+
+1. Defining Invariants:
+
+We will create a file that lists all the invariants we want to watch, with clear descriptions and how to get the data.
+These definitions will need approval from SIG Architecture and Testing Leads to make sure they are good and useful.
+
+- (TBD) Similar to existing conformance.yaml , it has to be part of test/e2e/invariants/data.yaml or similart
+
+2. Collecting Invariant Data:
+
+Our tests will collect the invariant data at the end of each run, using the methods we defined.
+We will only collect invariant data from places that are easily accessible.
+
+- (TBD) Ginkgo Reporting AfterSuite + Metrics Gatherer
+
+3. Storing and Processing Invariant Data:
+
+We will use existing tools to store the invariant data in a database that is easy to search.
```

Comment:
I don't think that's sufficient, we can't tell people "go pay to run bigquery to search for results".

This is why I'm still hesitant to abandon the approach of reporting a pass/fail, which we already disseminate in various forms including email alerts.

If we're going to commit to building a frontend for these results instead, we should be discussing that here.